timeboost-crypto/src/vess.rs

```diff
@@ -511,50 +511,73 @@ impl<C: CurveGroup> ShoupVess<C> {
 
         // k in S, then homomorphically shift commitment; k notin S, reproduce dealing from seed
         let subset_indices = self.map_subset_seed(subset_seed);
-        let mut subset_iter = subset_indices.iter().peekable();
-        let mut next_subset_idx = subset_iter.next();
-        for i in 0..self.num_repetition {
-            match next_subset_idx {
-                Some(j) if i == *j => {
+
+        // Convert subset_indices to a HashSet for O(1) lookup
+        let subset_set: std::collections::HashSet<usize> = subset_indices.iter().copied().collect();
+
+        // Create vectors to store the shifted polys and mre_cts data in the correct order
+        let mut shifted_polys_vec = Vec::new();
+        let mut mre_cts_vec = Vec::new();
+        let mut seeds_vec = Vec::new();
+
+        // Convert VecDeque to Vec to preserve order for subset items
+        for poly in shifted_polys.iter() {
+            shifted_polys_vec.push(poly.clone());
+        }
+        for ct in mre_cts.iter() {
+            mre_cts_vec.push(ct.clone());
+        }
+        for seed in seeds.iter() {
+            seeds_vec.push(*seed);
+        }
```

Comment:
Should be able to just do:

```suggestion
        let shifted_polys_vec: Vec<_> = shifted_polys.into_iter().collect();
        let mre_cts_vec: Vec<_> = mre_cts.into_iter().collect();
        let seeds_vec: Vec<_> = seeds.into_iter().collect();
```